### PR TITLE
feat: inserting document into existing design

### PR DESCRIPTION
**Pull request by RobPasMue**
Allows insertion of a document into an existing design. Seems to require some server side implementation. Pinging @b-matteo

### CODE REVIEW

**Review by AlejandroFernandezLuces — APPROVED** [review]
LGTM 🙂

File: src/ansys/geometry/core/designer/design.py [hunk 1]
```diff
@@ -581,6 +581,41 @@ def delete_beam_profile(self, beam_profile: Union[BeamProfile, str]) -> None:
...
+
+        # Insert the file into the design
+        self._design_stub.Insert(InsertRequest(filepath=filepath_server))
+        self._grpc_client.log.debug(f"File {file_location} successfully inserted into design.")
+
+        # Get a temporal design object to update the current one
+        tmp_design = Design("", self, read_existing_design=True)
+
+        # Update the design's components - add the new one
+        #
```
[the remark this answers is not in the document]
Comment (RobPasMue) [reply src/ansys/geometry/core/designer/design.py:607]: Nothing missing, just wanted to give some space

File: tests/integration/test_design_import.py [hunk 2]
```diff
@@ -240,3 +240,23 @@ def test_open_file(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
         # .prt
         design2 = modeler.open_file("./tests/integration/files/import/disk1.prt")
         assert len(design2.bodies) == 1
+
+
+def test_design_insert(modeler: Modeler):
+    """Test creation of a design, sketching a circle, extruding it and then inserting a
```
Comment (PipKat) [inline tests/integration/test_design_import.py:246]:
```suggestion
    """Test creation of a design, sketching a circle, extruding itn and then inserting a
```
If there is a summary table with method descriptions, this is never goign to fit!  LOL!

File: src/ansys/geometry/core/designer/design.py [hunk 3]
```diff
@@ -581,6 +581,45 @@ def delete_beam_profile(self, beam_profile: Union[BeamProfile, str]) -> None:
...
+    @check_input_types
+    @ensure_design_is_active
+    def insert_file(self, file_location: Union[Path, str]) -> None:
+        """
+        Insert a file into the design.
+
+        Parameters
+        ----------
+        file_location : Union[~pathlib.Path, str]
+            Location on disk to save the file to.
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/design.py:594]:
```suggestion
            Location on disk where the file is located.
```

File: src/ansys/geometry/core/designer/design.py [hunk 4]
```diff
@@ -581,6 +581,45 @@ def delete_beam_profile(self, beam_profile: Union[BeamProfile, str]) -> None:
...
+            self._components.extend(tmp_design.components)
+        else:
+            # Insert operation adds the inserted file as a component to the design.
+            for tmp_component in tmp_design.components:
+                # Otherwise, check which is the new component added
+                for component in self._components:
+                    if component.id == tmp_component.id:
+                        break
+                    # If not equal, add the component - since it has not been found
+                    self._components.append(tmp_component)
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/design.py:619]: I'll probably need to update the reference to parent component inside the new component

Comment (RobPasMue) [reply src/ansys/geometry/core/designer/design.py:628]: Done!

File: src/ansys/geometry/core/designer/design.py [hunk 5]
```diff
@@ -581,6 +581,57 @@ def delete_beam_profile(self, beam_profile: Union[BeamProfile, str]) -> None:
                 + " Ignoring request."
             )
 
+    @protect_grpc
+    @check_input_types
+    @ensure_design_is_active
+    def insert_file(self, file_location: Union[Path, str]) -> Component:
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/design.py:587]: Add 24.2 decorator

File: tests/integration/test_design_import.py [hunk 6]
```diff
@@ -240,3 +240,41 @@ def test_open_file(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
         # .prt
         design2 = modeler.open_file("./tests/integration/files/import/disk1.prt")
         assert len(design2.bodies) == 1
+
+
+def test_design_insert(modeler: Modeler):
+    """Test inserting a file into the design."""
```
Comment (RobPasMue) [inline tests/integration/test_design_import.py:246]: Skip on Linux

File: tests/integration/test_design_import.py [hunk 7]
```diff
@@ -240,3 +240,41 @@ def test_open_file(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
...
+    # Insert a different file
+    design.insert_file("./tests/integration/files/DuplicateFacesDesignBefore.scdocx")
+
+    # Check that there are two components
+    assert len(design.components) == 2
+    assert design.components[0].name == "Component_Cylinder"
+    assert design.components[1].name == "DuplicatesDesign"
+
+
+def test_design_insert_with_import(modeler: Modeler):
```
Comment (RobPasMue) [inline tests/integration/test_design_import.py:264]: Skip on Linux

**Pull request merged**